### ISSUE TITLE
Using an empty `master_authorized_networks` does not toggle control plane authorized networks

### ISSUE THREAD

**Issue by antoineco** [terraform-google-modules/terraform-google-kubernetes-engine#1638]
### TL;DR

Given a cluster where `enable_private_endpoint` is `false` and `master_authorized_networks` was set to a non empty value at any point in time (_Control plane authorized networks_ **Enabled**),

erasing the variable's value as demonstrated below does not toggle the _Control plane authorized networks_ feature.

`master_authorized_networks = []`
`master_authorized_networks = null`
`(no variable value)`

Instead, the module stops managing the feature entirely, and all the previous values remain set in the cluster's configuration.

### Expected behavior

An empty `master_authorized_networks` **disables** the _Control plane authorized networks_ feature (no access restriction on the control plane endpoint).

```diff
-            "master_authorized_networks_config": [
-              {
-                "cidr_blocks": [
-                  {
-                    "cidr_block": "10.60.0.0/17",
-                    "display_name": "VPC"
-                  },
-                ],
-                "gcp_public_cidrs_access_enabled": false
-              }
-            ],
+            "master_authorized_networks_config": [],
```

Alternatively, there could be a boolean variable controlling the feature.

Rationale:

- Disabling public access to the control plane endpoint is done with `enable_private_endpoint = true` already.
- Having _Control plane authorized networks_ Enabled, but without any authorized network, is equivalent to having `enable_private_endpoint = true` from a security standpoint.

### Observed behavior

- Terraform reports no change after the removal of the variable:
  ```
  No changes. Your infrastructure matches the configuration.
  ```

- The _Control plane authorized networks_ cluster feature is still enabled.

- All authorized networks previously defined are still effective:

![image](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/assets/3299086/acc9e7f2-5f1b-4f14-abbe-c6f79725213b)

### Terraform Configuration

```hcl
# From https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/5ef68525bdb1d6832fcd4ae2d6f81227a2cbd7bc/examples/simple_autopilot_private/main.tf#L53-L58

locals {
  cluster_type           = "simple-autopilot-private"
  network_name           = "simple-autopilot-private-network"
  subnet_name            = "simple-autopilot-private-subnet"
  master_auth_subnetwork = "simple-autopilot-private-master-subnet"
  pods_range_name        = "ip-range-pods-simple-autopilot-private"
  svc_range_name         = "ip-range-svc-simple-autopilot-private"
  subnet_names           = [for subnet_self_link in module.gcp-network.subnets_self_links : split("/", subnet_self_link)[length(split("/", subnet_self_link)) - 1]]
}


data "google_client_config" "default" {}

provider "kubernetes" {
  host                   = "https://${module.gke.endpoint}"
  token                  = data.google_client_config.default.access_token
  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
}

module "gke" {
  source                          = "../../modules/beta-autopilot-private-cluster/"
  project_id                      = var.project_id
  name                            = "${local.cluster_type}-cluster"
  regional                        = true
  region                          = var.region
  network                         = module.gcp-network.network_name
  subnetwork                      = local.subnet_names[index(module.gcp-network.subnets_names, local.subnet_name)]
  ip_range_pods                   = local.pods_range_name
  ip_range_services               = local.svc_range_name
  release_channel                 = "REGULAR"
  enable_vertical_pod_autoscaling = true
  enable_private_endpoint         = false
  enable_private_nodes            = true
  master_ipv4_cidr_block          = "172.16.0.0/28"
  network_tags                    = [local.cluster_type]

  # REMOVE ME
  master_authorized_networks = [
    {
      cidr_block   = "10.60.0.0/17"
      display_name = "VPC"
    },
  ]
}
```

### Terraform Version

```sh
Terraform v1.4.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.65.2
+ provider registry.terraform.io/hashicorp/google-beta v4.65.2
+ provider registry.terraform.io/hashicorp/kubernetes v2.20.0
+ provider registry.terraform.io/hashicorp/random v3.5.1
```

### Additional information

The workaround is to disable the feature manually throught the console or API. Since Terraform leaves it untouched, this doesn't cause any configuration drift during future `apply`'s.

**Comment by faganihajizada** [comment]
It would be nice to have option to set `gcp_public_cidrs_access_enabled` as `false` or `true`

**Comment by tanguynicolas** [comment]
Specifically, we'd like this configuration to be possible:

```hcl
enable_private_nodes          = true
deploy_using_private_endpoint = true
enable_private_endpoint       = true

master_authorized_networks = {
    gcp_public_cidrs_access_enabled = true  # True because it's mandatory
    
    cidr_blocks = {}  # None, because no need
}
```
